homeassistant/components/simplisafe/camera.py

```diff
@@ -0,0 +1,234 @@
+SERVICE_OC_CLIP = "capture_motion_clip"
+
+SERVICE_OC_IMAGE_SCHEMA = cv.make_entity_service_schema(
+    {
+        vol.Required(ATTR_ENTITY_ID): cv.entity_ids,
+        vol.Required(ATTR_FILENAME): cv.template,
+        vol.Optional(ATTR_WIDTH, default=720): vol.Coerce(int),
+    }
+)
+
+SERVICE_OC_CLIP_SCHEMA = cv.make_entity_service_schema(
+    {
+        vol.Required(ATTR_ENTITY_ID): cv.entity_ids,
+        vol.Required(ATTR_FILENAME): cv.template,
+    }
+)
+
+SERVICES = (
+    SERVICE_OC_IMAGE,
+    SERVICE_OC_CLIP,
+)
+
+
+def _write_image(to_file: str, image_data: bytes | None) -> None:
+    """Write image content to a file."""
+    if image_data is None:
+        return
+    os.makedirs(os.path.dirname(to_file), exist_ok=True)
+    with open(to_file, "wb") as img_file:
+        img_file.write(image_data)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up SimpliSafe cameras based on a config entry."""
+    simplisafe = hass.data[DOMAIN][entry.entry_id]
+
+    cameras: list[SimplisafeOutdoorCamera] = []
+
+    for system in simplisafe.systems.values():
+        if system.version == 2:
+            LOGGER.info("Skipping camera setup for V2 system: %s", system.system_id)
+            continue
+
+        cameras = [
+            SimplisafeOutdoorCamera(hass, simplisafe, system, camera)
+            for camera in system.cameras.values()
+            if camera.camera_type == CameraTypes.OUTDOOR_CAMERA
+        ]
+
+        async_add_entities(cameras)
+
+    platform = entity_platform.async_get_current_platform()
+
+    platform.async_register_entity_service(
+        SERVICE_OC_IMAGE,
+        SERVICE_OC_IMAGE_SCHEMA,
+        "async_save_image_handler",
+    )
+
+    platform.async_register_entity_service(
+        SERVICE_OC_CLIP,
+        SERVICE_OC_CLIP_SCHEMA,
+        "async_save_clip_handler",
+    )
+
+
+class SimplisafeOutdoorCamera(SimpliSafeEntity, Camera):
+    """A camera base class that supports motion capture media."""
+
+    _device: DeviceV3
+    _attr_image_last_updated: datetime | None = None
+    _attr_image_url: str | None = None
+    _attr_clip_url: str | None = None
+    _attr_hls_url: str | None = None
+    _attr_cached_image: bytes | None = None
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        simplisafe: SimpliSafe,
+        system: SystemV3,
+        camera: DeviceV3,
+    ) -> None:
+        """Initialize."""
+        SimpliSafeEntity.__init__(
+            self,
+            simplisafe,
+            system,
+            device=camera,
+            additional_websocket_events=[EVENT_CAMERA_MOTION_DETECTED],
+        )
+        Camera.__init__(self)
+
+        self._hass = hass
+
+    @property
+    def name(self) -> str | None:
+        """Return a good name for this camera."""
+        return self._device.name
+
+    @callback
+    def async_unload(self) -> None:
```

Comment:
What calls this method? Also, we should unregister the services when unloading the config entry, not when destroying a singe camera.